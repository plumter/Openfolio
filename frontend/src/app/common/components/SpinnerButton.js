import SVG from "react-inlinesvg";

const SpinnerButton = ({text, isLoading, ...props}) => {
    return  <button  
                {...props}
                disabled={props.disabled || isLoading}
            >
                {
                    isLoading ? 
                        <SVG
                            src="/assets/media/svg/spinner.svg"
                            className="inline-block splash-spinner stroke-white mx-auto h-6 w-6"
                        />
                        :
                        text
                }
            </button>
}

export default SpinnerButton;