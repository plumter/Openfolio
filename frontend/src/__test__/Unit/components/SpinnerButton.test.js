import SpinnerButton from 'app/common/components/SpinnerButton';
import { create } from 'react-test-renderer';



describe("Spinner Button Component Unit Testing", _ => {

    it('Test Spinner Button Cases',  () => {
      // 1st case
      const component = create(      
          <SpinnerButton 
              type="submit" 
              disabled={true} 
              className="w-full btn-primary" 
              isLoading={false}
              key={1}
          />,
        );
      expect(component.toJSON()).toMatchSnapshot();

      // 2nd case
      component.update(      
        <SpinnerButton 
            type="submit" 
            disabled={false} 
            className="w-full btn-primary" 
            isLoading={false}
            key={2}
        />,
      );
      expect(component.toJSON()).toMatchSnapshot();

      // 3rd case
      component.update(      
        <SpinnerButton 
            type="submit" 
            disabled={false} 
            className="w-full btn-primary" 
            isLoading={true}
            key={3}
        />,
      );
      expect(component.toJSON()).toMatchSnapshot();

      // 4th case
      component.update(      
        <SpinnerButton 
            type="submit" 
            disabled={true} 
            className="w-full btn-primary" 
            isLoading={true}
            key={4}
        />,
      );
      expect(component.toJSON()).toMatchSnapshot();
    });  
});