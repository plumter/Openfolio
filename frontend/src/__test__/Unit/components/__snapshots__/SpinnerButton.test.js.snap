// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Button Component Unit Testing Test Spinner Button Cases 1`] = `
<button
  className="w-full btn-primary"
  disabled={true}
  type="submit"
/>
`;

exports[`Spinner Button Component Unit Testing Test Spinner Button Cases 2`] = `
<button
  className="w-full btn-primary"
  disabled={false}
  type="submit"
/>
`;

exports[`Spinner Button Component Unit Testing Test Spinner Button Cases 3`] = `
<button
  className="w-full btn-primary"
  disabled={true}
  type="submit"
/>
`;

exports[`Spinner Button Component Unit Testing Test Spinner Button Cases 4`] = `
<button
  className="w-full btn-primary"
  disabled={true}
  type="submit"
/>
`;
